File: backend/src/modules/auth/models/user-model.ts
import mongoose, { Schema, type Document } from "mongoose";

export interface IUser extends Document {
  name: string;
  email: string;
  password: string;
  phone?: string;
  country?: string;
  state?: string;
  city?: string;
  role: "ADMIN" | "USER";
  refreshToken?: string;
  createdAt: Date;
  updatedAt: Date;
}

const userSchema = new Schema<IUser>(
  {
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true, lowercase: true },
    password: { type: String, required: true },
    phone: { type: String },
    country: { type: String },
    state: { type: String },
    city: { type: String },
    role: { type: String, enum: ["ADMIN", "USER"], default: "USER" },
    refreshToken: { type: String },
  },
  { timestamps: true },
);

export const User = mongoose.model<IUser>("User", userSchema);

File: backend/src/modules/auth/models/user_model.py
from __future__ import annotations

from beanie import Document
from pydantic import Field
from typing import Optional
from datetime import datetime


class User(Document):
    name: str
    email: str
    password: str
    phone: Optional[str] = None
    country: Optional[str] = None
    state: Optional[str] = None
    city: Optional[str] = None
    role: str = Field(default="USER", pattern=r"^(ADMIN|USER)$")
    profession: Optional[str] = None
    seniority: Optional[str] = Field(default=None, pattern=r"^(junior|mid-level|senior)$")
    age: Optional[int] = None
    total_days_active: int = 0
    language: str = Field(default="pt-BR", pattern=r"^(pt-BR|en-US)$")
    custom_cursor_enabled: bool = True
    total_seconds_active: int = 0
    refresh_token: Optional[str] = Field(default=None, alias="refreshToken")
    created_at: datetime = Field(default_factory=datetime.utcnow, alias="createdAt")
    updated_at: datetime = Field(default_factory=datetime.utcnow, alias="updatedAt")

    class Settings:
        name = "users"
        use_revision = False

    class Config:
        populate_by_name = True
